lib/streamlit/script_runner.py

```diff
@@ -353,16 +353,18 @@ def _run_script(self, rerun_data: RerunData) -> None:
         # in their previous script elements disappearing.
 
         try:
-            with source_util.open_python_file(self._session_data.script_path) as f:
+            with source_util.open_python_file(self._session_data.main_script_path) as f:
                 filebody = f.read()
 
             if config.get_option("runner.magicEnabled"):
-                filebody = magic.add_magic(filebody, self._session_data.script_path)
+                filebody = magic.add_magic(
+                    filebody, self._session_data.main_script_path
+                )
```

Comment:
Just curious if `script_path` should also be updated in the `add_magic` function def in `magic.py` (and consequently `magic_test.py`)?